cmd/otelcol/main.go

```diff
@@ -164,44 +156,72 @@ func checkRuntimeParams() {
 	// Check if memory ballast flag was passed
 	// If so, ensure memory ballast env var is not set
 	// Then set memory ballast and limit properly
-	ballastSize := getKeyValue(args, "--mem-ballast-size-mib")
+	ballastSize := getKeyValue(os.Args[1:], "--mem-ballast-size-mib")
 	if ballastSize != "" {
-		config = os.Getenv(ballastEnvVarName)
-		if config != "" {
-			log.Fatalf("Both %v and '--config' were specified, but only one is allowed", ballastEnvVarName)
+		if os.Getenv(ballastEnvVarName) != "" {
+			log.Fatalf("Both %v and '--mem-ballast-size-mib' were specified, but only one is allowed", ballastEnvVarName)
 		}
 		os.Setenv(ballastEnvVarName, ballastSize)
 	}
 	setMemoryBallast(memTotalSizeMiB)
 	setMemoryLimit(memTotalSizeMiB)
 }
 
-// Validate and set the configuration
-func setConfig() {
-	// Check if the config is specified via the env var.
-	config := os.Getenv(configEnvVarName)
-	// If not attempt to use a default config; supports Docker and local
-	if config == "" {
+// Validate and equate specified config file path flag to the config file path env var
+func setConfigSource() {
+	// Config file path from cmd flag --config.
+	pathFlag := getKeyValue(os.Args[1:], "--config")
+	// Config file path from env var.
+	pathVar := os.Getenv(configEnvVarName)
+	// Config YAML from env var.
+	yamlVar := os.Getenv(configYamlEnvVarName)
```

Comment:
I think it would be beneficial to add a corresponding `--config-yaml` flag/option.  We could treat `-` as a marker to read from stdin.

Reply:
Oh man, can I submit another PR for this. Also, please elaborate, "We could treat - as a marker to read from stdin". I don't quite follow.

Reply:
Yeah, something that can be added later.  `-` is often used as an instruction for the process to read a value from stdin `echo ok | vim -`